src/compiler/transformers/module/module.ts

```diff
@@ -871,12 +888,13 @@ namespace ts {
             const left = node.left;
             // If the left-hand-side of the binaryExpression is an identifier and its is export through export Specifier
             if (isIdentifier(left) && isAssignmentOperator(node.operatorToken.kind)) {
-               const sourceFile = getSourceFileOfNode(node);
-               const sourceFileName = sourceFile ? sourceFile.fileName : undefined;
-                if (bindingNameExportSpecifiers && bindingNameExportSpecifiers[sourceFileName] && hasProperty(bindingNameExportSpecifiers[sourceFileName], left.text)) {
+                if (bindingNameExportSpecifiersForFileMap &&
+                    bindingNameExportSpecifiersForFileMap[currentSourceFileIdDuringSubstitution] &&
+                    hasProperty(bindingNameExportSpecifiersForFileMap[currentSourceFileIdDuringSubstitution], left.text)) {
+
                     setNodeEmitFlags(node, NodeEmitFlags.NoSubstitution);
                     let nestedExportAssignment: BinaryExpression;
-                    for (const specifier of bindingNameExportSpecifiers[sourceFileName][left.text]) {
+                    for (const specifier of bindingNameExportSpecifiersForFileMap[currentSourceFileIdDuringSubstitution][left.text]) {

```

Comment:
As [above](https://github.com/Microsoft/TypeScript/pull/8991/commits/d5d2f8c19c69fb2da34019ce08cc987abe36b815#r66895073), use a local variable to hold this so we aren't performing the lookup a third time.